### ISSUE TITLE
SSH access only works after “activating” the connection on shell

### ISSUE THREAD

**Issue by kellertuer** [ForkIssues/Tracker#1453]
I noticed a small bug with fork, when working with SSH remotes.

For some remotes (It seems not GitHub, but my own gitea and gitlab) push/pull does not work directly.

The symptom is as follows:

1) Start Fork
2) Open a Repository (gitea / gitlab remote w/ ssh)
3) click push – it asks for a password three times and then ends with an error
4) open a terminal and type `ssh -T same-server-adress-as-for-the-repo-in-2` – this works and greets me with my username
5) change to Fork again, click again push – works

I just tried, it seems that this stays working even after closing/reopening fork; so it is maybe just the first connection to a server after restarting a Mac?

Let me know if/how I can provide more details here.

Edit: Affects Fork 2.9 and 2.10 on Mac OS (2.11 seems to not yet reach my update check, but it was also only released two days ago).

**Comment by DanPristupov** [comment]
Fork doesn't communicate with SSH or SSH-agent directly. Fork just calls `git` and the rest is made by git.

> 3. click push – it asks for a password three times and then ends with an error

Do you mean it asks the key passphrase?

> 4. open a terminal and type ssh -T same-server-adress-as-for-the-repo-in-2 – this works and greets me with my username

What happens if you run `git push` instead?

A few more questions:
- Do you have any checked key in the 'SSH Configuration' dialog in Fork? When a key is selected, Fork will override the default system SSH settings.
- What git instance do you use in Fork preferences?

**Comment by kellertuer** [comment]
> Fork doesn't communicate with SSH or SSH-agent directly. Fork just calls `git` and the rest is made by git.
> 
> > 1. click push – it asks for a password three times and then ends with an error
> 
> Do you mean it asks the key passphrase?

No, it asks for a password. Not for the passphrase. My local keys do not have a passphrase.

> 
> > 1. open a terminal and type ssh -T same-server-adress-as-for-the-repo-in-2 – this works and greets me with my username
> 
> What happens if you run `git push` instead?

Same for fetch and push.

> 
> A few more questions:
> 
> * Do you have any checked key in the 'SSH Configuration' dialog in Fork? When a key is selected, Fork will override the default system SSH settings.

I never used that; I just use about 6 keys and a properly configured `.ssh/config` – also on terminal this error does not occur
But I looked where it is and currently it seems only the one from GitHub is checked, the others are not. May that be the cause? Why might this one be checked? I really never was in this dialog before. How did my other keys end up in there? Does Fork read my `.ssh/config`? Maybe checking them actively there might fix this problem here?

But then I would really like to understand how all 6 ended up in that list but just one is checked, when I was never in that dialog nor have ever noticed that it exists.

> * What git instance do you use in Fork preferences?

Continuing from the last point, fork uses ”2.31.1 For instance”, I never changed that. So one difference to the statement that it works fine on terminal is, that on terminal I have the Mac OS system git
```
❯ git --version
git version 2.30.1 (Apple Git-130)
````

**Comment by DanPristupov** [comment]
> Continuing from the last point, fork uses ”2.31.1 For instance”, I never changed that. So one difference to the statement that it works fine on terminal is, that on terminal I have the Mac OS system git

Try to switch Fork to the system git instance. This way it will be same configuration as you have in terminal.

**Comment by DanPristupov** [comment]
> I never used that; I just use about 6 keys and a properly configured .ssh/config – also on terminal this error does not occur
But I looked where it is and currently it seems only the one from GitHub is checked, the others are not. 

Previously those settings looked differently and was part of the GitHub Account view.

> May that be the cause? 

Yes, that's the cause. I think you need to uncheck all keys to make Fork use the default system settings.

**Comment by kellertuer** [comment]
Thanks, I will switch.

And – it is a little strange to uncheck them so they work correctly, especially since the only one that _is_ checked (the one I use for GitHub) seems to be the one that worked until now? Well, maybe using the system git together with unchecking all will solve this. I will report.

**Comment by DanPristupov** [comment]
So,

1. Uncheck all SSH keys in the SSH configuration dialog to ensure that the default system SSH configuration is used.
2. Try both git instances (Fork and the system one) in Fork Preferences.
